CMakeLists.txt

```diff
@@ -286,6 +286,8 @@ endif()
 
 # @Intent: Set compiler specific macro inclusion of AUTOCONF_H
 zephyr_compile_options("SHELL: $<TARGET_PROPERTY:compiler,imacros> ${AUTOCONF_H}")
+# @Intent: Set compiler specific macro inclusion of VERSION_H
+zephyr_compile_options("SHELL: $<TARGET_PROPERTY:compiler,imacros> ${VERSION_H}")
```

Comment:
OK with change except for this, should remain an include file which users #include if they need it. Having it in kernel.h I am undecided upon as it seems likely that some codebase or library somewhere will have a conflicting define

Reply:
This line is the essence of the second way. Do you think name spacing the macros in the `version.h` (i.e. `ZEPHYR_*) would help with mitigating the potential conflict?